rai/client.go

```diff
@@ -426,19 +438,28 @@ func parseHttpResponse(rsp *http.Response) (interface{}, error) {
 	}
 
 	mediaType, params, _ := mime.ParseMediaType(rsp.Header.Get("Content-Type"))
+
 	if mediaType == "application/json" {
 		return data, nil
 	} else if mediaType == "multipart/form-data" {
 		boundary := params["boundary"]
 		return parseMultipartResponse(data, boundary)
+	} else if mediaType == "application/x-protobuf" {
+		var metadataInfo generated.MetadataInfo
+		err := proto.Unmarshal(data, &metadataInfo)
```

Comment:
Again not sure I understand Go ^^ but we seem not to be using our function `unmarshal` that is defined above.

Reply:
Ah, I see, `unmarshal` is called in `request`. But before that we use this method `parseHTTPResponse` and in here we actually unmarshal the Proto and above in `unmarshal` we simply pass it through :)